ignite/metrics/precision.py

```diff
@@ -15,28 +15,39 @@ class _BasePrecisionRecall(_BaseClassification):
     def __init__(
         self,
         output_transform: Callable = lambda x: x,
-        average: bool = False,
+        average: Union[bool, str] = False,
         is_multilabel: bool = False,
         device: Union[str, torch.device] = torch.device("cpu"),
     ):
 
+        if type(average) != bool and average not in ["micro", "weighted", "samples"]:
```

Comment:
How about doing this like that ?
```suggestion
        if not (isinstance(average, bool) or average in ["micro", "weighted", "samples"]):
```